tests/test_datacollector_mutable_fix.py

```diff
@@ -0,0 +1,49 @@
+"""Test DataCollector correctly deep-copies mutable agent data."""
+
+import unittest
```

Comment:
Can you rewrite these tests to use pytest and include them in the proper file that contains the tests for the data collector?